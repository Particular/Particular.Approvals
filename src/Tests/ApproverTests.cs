namespace Tests
{
    using NUnit.Framework;
    using Particular.Approvals;

    [TestFixture]
    class ApproverTests
    {
        [Test]
        public void Can_Approve_Text()
        {
            Approver.Verify("Text to\r\napprove");
        }

        [Test]
        public void Can_Approve_Text_With_Category()
        {
            Approver.Verify("Text to\r\napprove", "test");
        }

        [Test]
        public void Can_Approve_Object()
        {
            var sample = new Sample { Value1 = "Value", Value2 = 42 };

            Approver.Verify(sample);
        }

        [Test]
        public void Can_Approve_Object_With_Category()
        {
            var sample = new Sample { Value1 = "Value", Value2 = 42 };

            Approver.Verify(sample, "test");
        }
    }

    class Sample
    {
        public string Value1 { get; set; }

        public int Value2 { get; set; }
    }
}
