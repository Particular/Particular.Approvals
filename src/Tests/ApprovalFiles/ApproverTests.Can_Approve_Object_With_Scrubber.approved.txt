{
  "Value1": "Value",
  "Value2": 100
}